darshan-test/python_runtime_tests.py

```diff
@@ -175,3 +174,50 @@ def test_env_vars_config(tmpdir):
         report = darshan.DarshanReport(path_to_log)
         # NOTE: could eventually do some more sanity
         # checks on the report data here
+
+def test_forked_process(tmpdir):
+    # regression test for gh-786
+    root_path = os.environ.get("DARSHAN_ROOT_PATH")
+    darshan_install_path = os.environ.get("DARSHAN_INSTALL_PATH")
+    test_script_path = os.path.join(root_path,
+                                    "darshan-test",
+                                    "python_mpi_scripts",
+                                    "runtime_prog_issue_786.py")
+    darshan_lib_path = os.path.join(darshan_install_path,
+                                    "lib",
+                                    "libdarshan.so")
+
+    with tmpdir.as_cwd():
+        cwd = os.getcwd()
+        subprocess.check_output([
+                     "python",
```

Comment:
I needed this to be changed to `python3` to execute locally. That would be my normal expectation on Ubuntu shell call-outs with Python anyway, though CI seems happy enough.

I think rather than asking you to change that now, what I'll do is open a separate small issue suggesting we try something a bit more portable like asking Python where its executable is directly:

```python
import sys
py_exe = sys.executable
```